### PR TITLE
Fix #5303: Use emoji flags for Android 5.0+ users / default to the Gravatar option for other users

### LINKED ISSUE #5303: Country Flags Missing in Apps Stats
The Australian and Serbian flags don't show up in the app stats. This might eventually be fixed by using flag emojis, instead of images, as mentioned in #5206. But that issue's priority is low, so I'm creating this one, as well.

image

WordPress – 6.6
HUAWEI RIO-L02
android – 5.1
English

### PR DESCRIPTION
**Pull request by maxme**
Pros:
* 257 countries supported
* no network requests
* no extra resources

Cons:
* waves
* only works in Android >= 5.0

screenshot-2017-03-30_10 53 39 431

Android < 5.0 usage is pretty low (~14%), I think it's fair to not fix it for them.

cc @daniloercoli

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUtils.java [hunk 1]
```diff
@@ -521,4 +521,10 @@ public static String getSinceLabel(Context ctx, String dataSubscribed) {
 
         return "";
     }
+
+    public static String countryCodeToEmoji(String countryCode) {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUtils.java]: Do we have some doc about this routine? Even adding al link should be ok.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/StatsGeoviewsFragment.java [hunk 2]
```diff
@@ -250,16 +251,21 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
             holder.setEntryText(entry);
-            holder.totalsTextView.setText(FormatUtils.formatDecimal(total));
 
-            // image (country flag)
-            holder.networkImageView.setImageUrl(
-                    GravatarUtils.fixGravatarUrl(imageUrl, mResourceVars.headerAvatarSizePx),
-                    WPNetworkImageView.ImageType.BLAVATAR);
-            holder.networkImageView.setVisibility(View.VISIBLE);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                // On Android >= 5.0, ese the emoji flag
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/StatsGeoviewsFragment.java]: It seems there is a typo here.

**Comment by daniloercoli** [comment]
Thanks @maxme for taking care of this issue on Stats. 
I'm fine on having this PR merged in the app, even though I don't like the "wave" effect.

I also want to ping @bysl  on this, since the new flags are slightly different from the ones we have in Calypso. Susan what are your thoughts on this? 
screen shot 2017-03-30 at 11 42 42

**Comment by maxme** [comment]
> I'm fine on having this PR merged in the app, even though I don't like the "wave" effect.

Unfortunately there is no other solution that would not require extra amount of work (and package 257 icons or 3 times this if we use PNGs).

File: WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUtils.java (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/StatsUtils.java]: Good call, fixed in  	9e0df80

**Comment by daniloercoli** [comment]
> Unfortunately there is no other solution that would not require extra amount of work (and package 257 icons or 3 times this if we use PNGs).

I'm OK with merging this PR. @bysl Are you ok with the new icons being "waved"?

**Comment by mattmiklic** [comment]
> Unfortunately there is no other solution that would not require extra amount of work (and package 257 icons or 3 times this if we use PNGs).

@elibud used the Twemoji set of flags when this was updated recently on iOS. Could we do the same on Android? https://github.com/wordpress-mobile/WordPressCom-Stats-iOS/pull/440

**Comment by maxme** [comment]
> Twemoji set of flags when this was updated recently on iOS. Could we do the same on Android? wordpress-mobile/WordPressCom-Stats-iOS#440

We could, but we would have to either:

a. Convert the SVGs to vector drawables. Actually this is the first thing I tried. I stumbled upon complicated flags that uses some SVGs specifics that are not supported by vector drawables. Example of flags that fails conversion to vector drawable: https://github.com/twitter/twemoji/blob/gh-pages/2/svg/1f1e6-1f1e8.svg and https://github.com/twitter/twemoji/blob/gh-pages/2/svg/1f1e6-1f1f1.svg - so in that case we need to manually redraw these flags to convert them in vector drawables (using  SVGs path only). Or we can:
b. Convert the SVGs to PNGs, but we need to support 3 resolutions, so we need to package 257 * 3 = 771 icons, which seems totally unnecessary when we can just use the emojis.

**Comment by bysl** [comment]
Looking at the current alternatives, using emoji flags seems like the best solution for now.

**Comment by mattmiklic** [comment]
> Convert the SVGs to vector drawables. Actually this is the first thing I tried. I stumbled upon complicated flags that uses some SVGs specifics that are not supported by vector drawables. Example of flags that fails conversion to vector drawable

I may have found a solution that's not too time-intensive; I was able to successfully convert both of these SVGs to Android vector drawables by re-saving the SVG in Illustrator first before doing the conversion to Android vector drawable.

screen shot 2017-03-30 at 12 56 51 pm
screen shot 2017-03-30 at 2 46 15 pm

At a quick glance they both appear to be faithful to the originals (and don't have the errors I saw when converting directly from the Twemoji source). I can see how long it would take for me to convert them during my groundskeeping time on Monday. It's a small thing, but the Twemoji flags are really a lot better than the wavy flags, on either platform, so I think it's worth it if it won't take too much time to implement them.

**Comment by mattmiklic** [comment]
Let's go ahead and ship this with the standard wavy emoji flags. I'd like to improve these by switching to Twemoji when there's time, but there's no sense in letting it hold up this PR. LGTM as-is. 👍

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**